### PR TITLE
Fixed non-realistic upper bound bug, Added normal and edge case tests and a minor performance optimisation.

### PR DESCRIPTION
**Pull request by Axibord**
- Fixed an edge case with an upper bound of 0.
- Added normal and edge case tests.
- added a contribution section to the README.md

### CODE REVIEW

File: chaos_test.go [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+	"github.com/stretchr/testify/require"
+)
+
+func TestInt(t *testing.T) {
+	t.Run("deterministic output", func(t *testing.T) {
+		assert.Equal(t, Int(100, "seed"), Int(100, "seed"))
+	})
+
+	t.Run("different seeds produce different results", func(t *testing.T) {
+		assert.NotEqual(t, Int(100, "seed1"), Int(100, "seed2"))
```
Comment (raphoester) [inline chaos_test.go]: what about collisions ? there is 1% chance that this test fails

File: chaos_test.go [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+func TestBool(t *testing.T) {
+	t.Run("deterministic output", func(t *testing.T) {
+		assert.Equal(t, Bool("seed"), Bool("seed"))
+	})
+
+	t.Run("different seeds produce different results", func(t *testing.T) {
+		found := false
+		for i := 0; i < 100 && !found; i++ {
+			if Bool(i) != Bool(i+1) {
+				found = true
```
Comment (raphoester) [inline chaos_test.go]: break

File: chaos_test.go [hunk 3]
```diff
@@ -0,0 +1,289 @@
+package chaos
```
Comment (raphoester) [inline chaos_test.go]: prefer blackbox tests (`package chaos_test`)

File: chaos_test.go [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+		result2, err2 := UniqueSliceItems(items, 3, "seed")
+		require.NoError(t, err1)
+		require.NoError(t, err2)
+		assert.Equal(t, result1, result2)
+	})
+
+	t.Run("returns error when not enough items", func(t *testing.T) {
+		items := []int{1, 2, 3}
+		_, err := UniqueSliceItems(items, 4, "seed")
+		assert.Error(t, err)
```
Comment (raphoester) [inline chaos_test.go]: `assert.ErrorIs()` ?

File: chaos.go [hunk 5]
```diff
@@ -9,11 +9,20 @@ import (
 	"golang.org/x/exp/constraints"
 )
 
-// Int returns a deterministic number between 0 and n included.
+// Int generates a deterministic integer between 0 and n (inclusive) based on the provided seed.
+// Behavior:
+//   - If n <= 0, the function returns 0.
+//   - The function uses the seed to initialize a random number generator,
```
Comment (raphoester) [inline chaos.go]: this behavior is package wide and i feel like it's not useful to specify on each function's doc

File: chaos_test.go (hunk 1 above)
Comment (Axibord) [reply chaos_test.go]: > what about collisions ? there is 1% chance that this test fails

Hmm...you're right.

What about iterating 1000 times? We'll get like (1/101)^1000 chance of collision, which is astronomically small.
I think it's good for now.

If yes, should we update all other functions with the same problem?

```
const iterations = 1000
found := false
for i := 0; i < iterations && !found; i++ {
	if Int(100, "seed1") != Int(100, "seed2") {
		found = true
	}
}
```

File: chaos_test.go (hunk 4 above)
Comment (Axibord) [reply chaos_test.go]: Using `assert.ErrorIs()` wouldn't be appropriate here because there's no specific error type to compare against.

We could define a custom error type in the chaos package and use that instead of `fmt.Errorf()` 
But I think we just keep it as it is for now.

**Comment by Axibord** [comment]
@raphoester Fixed all the comments, but I see there is a huge change to the package all of a sudden 😅 I guess it's better to close this PR.

**Comment by raphoester** [comment]
I'm adapting everything dw ^^

**Comment by Axibord** [comment]
> I'm adapting everything dw ^^

Very cool, thanks!

**Comment by raphoester** [comment]
waiting for your opinion @Axibord , I think we're good now

**Comment by Axibord** [comment]
> waiting for your opinion @Axibord , I think we're good now

I think it's pretty good for now, we can improve it after if it's necessary.

I don't know what to think about the complexity introduced by the new pattern, I'm a bit skeptical, but it's a choice.

I'm merging the PR. Let's go :)

**Comment by Axibord** [comment]
> > waiting for your opinion @Axibord , I think we're good now
> 
> I think it's pretty good for now, we can improve it after if it's necessary.
> 
> I don't know what to think about the complexity introduced by the new pattern, I'm a bit skeptical, but it's a choice.
> 
> I'm merging the PR. Let's go :)

Oups can't merge, go ahead then.

**Comment by raphoester** [comment]
I had to make this arbitrary choice because the repeated "seed" argument was polluting everything. 
From the caller perspective it's still very understandable. 
I have to admit that the copy paste of methods/functions is ugly but that part could be fixed later through some code generation, so i guess it's fine for now
